crypto/ocsp/ocsp_vfy.c

```diff
@@ -347,87 +360,45 @@ static int ocsp_check_delegated(X509 *x)
 }
 
 /*
- * Verify an OCSP request. This is fortunately much easier than OCSP response
- * verify. Just find the signers certificate and verify it against a given
- * trust value.
+ * Verify an OCSP request. This is much easier than OCSP response verify.
+ * Just find the signer's certificate and verify it against a given trust value.
+ * Returns 1 on success, 0 on error, or -1 on fatal error like malloc failure.
```

Comment:
Correct that the result values of this public (but undocumented) function changes.
So I've added a small entry to `CHANGES.md`.

Reply:
Hmm, I am not sure it is that easy. This kind of behaviour change is really bad as the caller can mistreat the fatal error for success and there is no easy indication that such API contract change happened. Even though the API call was undocumented, you could easily find from the implementation that only 0 and 1 are valid return values. If we did not have any API calls with strictly boolean return values, it would be a different matter but that is not the case.

Reply:
For example searchcode.com reveals that Ruby already depends on this boolean return value.

Reply:
Agreed - changing the return value in this way could easily break code - so I don't think this change can be made without an OMC exception (which I doubt would be granted for this case).

Reply:
All right, so made sure that `OCSP_request_verify()` still returns 0 on fatal errors.
Consequently reverted the addition to `CHANGES.md`.